apps/web/pages/api/event-types/[slug].ts

```diff
@@ -0,0 +1,52 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import z from "zod";
+
+import { getServerSession } from "@calcom/features/auth/lib/getServerSession";
+import { checkRateLimitAndThrowError } from "@calcom/lib/checkRateLimitAndThrowError";
+import getIP from "@calcom/lib/getIP";
+import { HttpError } from "@calcom/lib/http-error";
+import prisma from "@calcom/prisma";
+
+const querySchema = z.object({
+  slug: z.string({ required_error: "event-type slug is required" }),
+});
+
+async function handler(req: NextApiRequest & { userId?: number }, res: NextApiResponse) {
+  const userIp = getIP(req);
+
+  await checkRateLimitAndThrowError({
+    rateLimitingType: "core",
+    identifier: userIp,
+  });
+
+  const session = await getServerSession({ req, res });
+  if (!session) return res.status(409).json({ message: "Unauthorized" });
+
+  const parsedQuery = querySchema.safeParse(req.query);
+  if (!parsedQuery.success) throw new HttpError({ statusCode: 400, message: parsedQuery.error.message });
+
+  const {
+    data: { slug },
+  } = parsedQuery;
+  if (!slug) return res.status(400).json({ message: "event-type slug is needed" });
+
+  const eventType = await prisma.eventType.findFirst({
```

Comment:
This only applies to event-types "owned" by whoever is accessing the api.
I saw multiple users may be allowed access to the same event-type but not sure
if that should be enforced in this endpoint

For instance I initially had something to check if user is allowed access to the matched event type
```ts
  const canAccessEventType = eventType.users.some((user) => user.id === session.user.id);
  if (!canAccessEventType) return res.status(409).json({ message: "Unauthorized" });
```

Had to remove it.

Also I just returned a minimal payload but that can be updated as needed.